packages/graph/hash_graph/lib/graph/src/store/postgres/query/mod.rs

```diff
@@ -0,0 +1,46 @@
+#![expect(dead_code, reason = "Work in progress")]
+
+//! Postgres implementation to compile queries.
+
+pub mod database;
+
+use std::fmt::{self, Formatter};
+
+use crate::store::{
+    postgres::query::database::{Column, Table},
+    query::QueryRecord,
+};
+
+/// A structural query, which can be compiled into a statement in Postgres.
+// TODO: Implement for `ReadQuery<DataType>`, `ReadQuery<PropertyType>`, etc. when associated types
+//       are implemented
+pub trait Query {
+    type Field: Field;
+    type Record: QueryRecord;
+
+    /// The table used for this `Query`.
+    fn base_table() -> Table;
+}
+
+/// A field used in queries.
+// TODO: Implement for `DataTypeField`, `PropertyTypeQueryField`, etc. (not added yet)
+pub trait Field {
+    /// The [`Column`], where this field lives in.
+    fn column(&self) -> Column;
+}
+
+/// A path used in queries.
+// TODO: Implement for `DataTypeQueryPath`, `PropertyTypeQueryPath`, etc. (not added yet)
+pub trait Path {
+    /// Returns a list of tables required to traverse this path.
+    fn tables(&self) -> Vec<Table>;
+
+    /// Returns the [`Column`] where the path ends at.
+    fn column(&self) -> Column;
+}
+
+/// Renders the object into a Postgres compatible format.
+pub trait Render {
```

Comment:
I agree with not using `Display` or `Debug`, I'm not really sure about "rendering" though. Should we maybe just call this `TranspileToSql` or something?

Reply:
Sounds good

Reply:
I changed this to `Transpile` in 0edb016c377d73120eeeb41e2f242e1dc168b391 (This is local to the `postgres` module, so I don't think it's worth writing `to_sql` each time as I use it a lot)